src/main/java/carpet/script/utils/ShapeDispatcher.java

```diff
@@ -546,6 +563,129 @@ public long calcKey()
         }
     }
 
+    public static class DisplayedSprite extends ExpiringShape {
+        private final Set<String> required = Set.of("pos");
+        private final Map<String, Value> optional = Map.ofEntries(
+                entry("facing", new StringValue("player")),
+                entry("tilt", new NumericValue(0)),
+                entry("lean", new NumericValue(0)),
+                entry("turn", new NumericValue(0)),
+                entry("height", new NumericValue(1)),
+                entry("width", new NumericValue(1)),
+                entry("obj_size", new NumericValue(1)),
+                entry("light_fromblock", new NumericValue(-999)),
+                entry("light_fromsky", new NumericValue(-999)),
+                entry("needf3b", BooleanValue.FALSE));
+        private boolean isitem;
+
+        @Override
+        protected Set<String> requiredParams() {
+            return Sets.union(Sets.union(super.requiredParams(), required), Set.of(isitem ? "item" : "block"));
+        }
+
+        @Override
+        protected Set<String> optionalParams() {
+            return Sets.union(Sets.union(super.optionalParams(), optional.keySet()),isitem ? Set.of("item_transform_type") : Set.of());
+        }
+
+        public DisplayedSprite(boolean i) {
+            isitem = i;
+        }
+
+        Vec3 pos;
+
+        Direction facing;
+
+        float tilt;
+        float lean;
+        float turn;
+        float size;
+        int light_fromblock;
+        int light_fromsky;
+
+        float height;
+        float width;
+        boolean needf3b;
+        CompoundTag blockEntity;
+        BlockState blockState;
+        ItemStack item = null;
+        TransformType item_transform_type;
+
+        @Override
+        protected void init(Map<String, Value> options) {
+            super.init(options);
+            pos = vecFromValue(options.get("pos"));
+            blockState = ((BlockValue) options.getOrDefault("block", BlockValue.AIR)).getBlockState();
+            blockEntity = ((BlockValue) options.getOrDefault("block", BlockValue.AIR)).getData();
+            NBTSerializableValue item_ = (NBTSerializableValue) options.getOrDefault("item", null);
+            if (item_ != null) {
+                this.item = ItemStack.of(item_.getCompoundTag());
+            }
+            light_fromblock = NumericValue
+                    .asNumber(options.getOrDefault("light_fromblock", optional.get("light_fromblock"))).getInt();
+            if (light_fromblock > 15)
+                light_fromblock = 15;
+            light_fromsky = NumericValue.asNumber(options.getOrDefault("light_fromsky", optional.get("light_fromsky")))
+                    .getInt();
+            if (light_fromsky > 15)
+                light_fromsky = 15;
+
+            item_transform_type = TransformType.valueOf(options.getOrDefault("item_transform_type",new StringValue("GUI")).getString());
```

Comment:
this is typically not specified, so make a shortcut rather than creating a custom objects.

Reply:
i did not get what you mean.....
what object is the custom objects you said?

Reply:
i fixed it in my understanding